File: delivery/go-sdk/models/receiver/entity/delivery/cancel_entity.go
/**
 * @Author Hanqiang
 * @Date 2025/12/9
 * @PackageName: entity
 * @ClassName: cancel_entity
 * @Description: 取消配送实体定义
 * @Version 1.0
 */

package entity

// CancelReq 取消配送请求参数
// command: cancel
// 必接: 是
// 说明: 麦芽田平台调用三方配送服务取消配送订单
type CancelReq struct {
	// SourceOrderNo 麦芽田侧运单号
	// 麦芽田平台的订单号
	SourceOrderNo string `json:"source_order_no"`

	// SourceDeliveryNo 三方配送单号
	// 配送服务商的订单号
	SourceDeliveryNo string `json:"source_delivery_no"`

	// CancelReasonType 取消类型
	// 1: 用户取消
	// 2: 商户取消
	// 3: 客服取消
	// 4: 系统取消
	// 5: 其他原因
	// 详见取消类型枚举
	CancelReasonType int `json:"cancel_reason_type"`

	// CancelReason 取消原因
	// 取消订单的具体原因描述
	CancelReason string `json:"cancel_reason"`
}

// CancelResp 取消配送响应参数
type CancelResp struct {
	// Status 处理状态
	// 1: 成功; 0: 失败
	Status int `json:"status,default=1"`

	// ErrorCode 错误码
	// 取消失败时的错误码
	ErrorCode string `json:"error_code,optional"`

	// Reason 失败原因
	// 取消失败时的原因描述
	Reason string `json:"reason,optional"`

	// CancelAmount 取消违约金
	// 单位：分
	CancelAmount int64 `json:"cancel_amount,optional"`
}

File: delivery/java-sdk/src/main/java/com/maiyatian/delivery/sdk/models/receiver/entity/delivery/CancelEntity.java
/*
 * 麦芽田配送开放平台 Java SDK
 * Copyright (c) 2025 麦芽田开放平台
 * MIT License
 */

package com.maiyatian.delivery.sdk.models.receiver.entity.delivery;

/**
 * 取消配送实体
 * <p>
 * 用于麦芽田请求三方服务取消配送，command值为cancel，是必接接口
 * 
 * @author SDK Generator
 * @version 1.0.0
 * @since 1.0.0
 */
public class CancelEntity {
    
    /**
     * 取消配送请求参数
     * <p>
     * command: cancel
     * 必接: 是
     * 说明: 麦芽田平台调用三方配送服务取消配送订单
     */
    public static class Req {
        
        /**
         * 麦芽田侧运单号
         * <p>麦芽田平台的订单号
         */
        private String sourceOrderNo;
        
        /**
         * 三方配送单号
         * <p>配送服务商的订单号
         */
        private String sourceDeliveryNo;
        
        /**
         * 取消类型
         * <ul>
         * <li>1: 用户取消</li>
         * <li>2: 商户取消</li>
         * <li>3: 客服取消</li>
         * <li>4: 系统取消</li>
         * <li>5: 其他原因</li>
         * </ul>
         * 详见取消类型枚举
         */
        private int cancelReasonType;
        
        /**
         * 取消原因
         * <p>取消订单的具体原因描述
         */
        private String cancelReason;
        
        // ==================== Getter 和 Setter 方法 ====================
        
        public String getSourceOrderNo() {
            return sourceOrderNo;
        }
        
        public void setSourceOrderNo(String sourceOrderNo) {
            this.sourceOrderNo = sourceOrderNo;
        }
        
        public String getSourceDeliveryNo() {
            return sourceDeliveryNo;
        }
        
        public void setSourceDeliveryNo(String sourceDeliveryNo) {
            this.sourceDeliveryNo = sourceDeliveryNo;
        }
        
        public int getCancelReasonType() {
            return cancelReasonType;
        }
        
        public void setCancelReasonType(int cancelReasonType) {
            this.cancelReasonType = cancelReasonType;
        }
        
        public String getCancelReason() {
            return cancelReason;
        }
        
        public void setCancelReason(String cancelReason) {
            this.cancelReason = cancelReason;
        }
    }
    
    /**
     * 取消配送响应参数
     */
    public static class Resp {
        
        /**
         * 处理状态
         * <p>1: 成功; 0: 失败
         */
        private int status = 1;
        
        /**
         * 错误码
         * <p>取消失败时的错误码
         */
        private String errorCode;
        
        /**
         * 失败原因
         * <p>取消失败时的原因描述
         */
        private String reason;
        
        /**
         * 取消违约金
         * <p>单位：分
         */
        private long cancelAmount;
        
        // ==================== Getter 和 Setter 方法 ====================
        
        public int getStatus() {
            return status;
        }
        
        public void setStatus(int status) {
            this.status = status;
        }
        
        public String getErrorCode() {
            return errorCode;
        }
        
        public void setErrorCode(String errorCode) {
            this.errorCode = errorCode;
        }
        
        public String getReason() {
            return reason;
        }
        
        public void setReason(String reason) {
            this.reason = reason;
        }
        
        public long getCancelAmount() {
            return cancelAmount;
        }
        
        public void setCancelAmount(long cancelAmount) {
            this.cancelAmount = cancelAmount;
        }
    }
}